roc_jni/src/main/cpp/endpoint.cpp

```diff
@@ -0,0 +1,211 @@
+#include "org_rocstreaming_roctoolkit_Endpoint.h"
+#include "common.h"
+#include "endpoint.h"
+
+#include <roc/config.h>
+
+#define ENDPOINT_CLASS              PACKAGE_BASE_NAME "/Endpoint"
+#define PROTOCOL_CLASS              PACKAGE_BASE_NAME "/Protocol"
+
+int endpoint_unmarshal(JNIEnv *env, roc_endpoint** endpoint, jobject jendpoint) {
```

Comment:
Let's update endpoint_unmarshal() to provide the following contract:

* at start, it makes an assertion that endpoint is NULL
* if it fails, it guarantees that endpoint is deallocated and is set to NULL
* if it succeeds, it guarantees that endpoint is allocated and non-NULL

Otherwise, it's hard to determine on the caller side, whether endpoint was already allocated and deallocated.